### DISCUSSION TITLE
How to make a project as standalone project and as a part of a large project?

### DISCUSSION THREAD

**Discussion by taozuhong** [mesonbuild/meson#11803] [General]
Hi all,

I have a project build with Meson, it contains 506 compile objects, split into three libraries plus one app, now it cost a lot of time to compile, so I want to:
1. Make every library could build itself standalone.
2. It could be as a part of a top-level project, like I18N handle.

How to implement it?
Thanks in advance.

**Comment by xclaesse** [comment]
You can use `ninja -C <builddir> path/to/libfoo.so ` to build only the given target and its dependencies. If your libraries can be standalone projects in theirself, you could use subprojects: In the main project source tree you move those libs into `subprojects/mylib1/` and the main project can then use `dependency('mylib1')`. That way you can also only configure/compile mylib1 subdir separately. https://mesonbuild.com/Subprojects.html
